pkg/platform/yaml/yaml.go

```diff
@@ -358,14 +362,33 @@ func (ys *Yaml) WriteMock(mockRead platform.KindSpecifier, ctx context.Context)
 	// 	mock.Name = "mocks"
 	// }
 
-	err = ys.Write(ys.MockPath, "mocks", mockYaml)
+	err = ys.Write(ys.MockPath, "mocks", mockYaml, false)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func (ys *Yaml) RewriteMocks(mocks []*models.Mock, testSet string) error {
+	ys.MockPath = filepath.Join(ys.MockPath, testSet)
+	for i, mock := range mocks {
+		mockYaml, err := EncodeMock(mock, ys.Logger)
+		if err != nil {
+			return err
+		}
+		if i == 0 {
+			err = ys.Write(ys.MockPath, "mocks", mockYaml, true)
+		} else {
+			err = ys.Write(ys.MockPath, "mocks", mockYaml, false)
+		}
+		if err != nil {
+			return err
```

Comment:
instead of this can the whole file be deleted before hand ? and so that write function isn't refactored and this is a very specific case of just checking for a single element of the array.

Reply:
shall I change update it to use the delete file logic ?

Reply:
Would be a better option, we already have all the mocks we need we can simply delete it because in truncate that's the operation we're doing.

Reply:
done!